modules/openapi-json-schema-generator/src/main/resources/python/endpoint.handlebars

```diff
@@ -386,8 +386,20 @@ _all_accept_content_types = (
 
 class BaseApi(api_client.Api):
 
+    @typing.overload
     def _{{operationId}}_oapg(
-    {{> endpoint_args selfType="api_client.Api" }}
+    {{> endpoint_args isOverload=true skipDeserialization="False"}}
+
+    @typing.overload
+    def _{{operationId}}_oapg(
+    {{> endpoint_args isOverload=true skipDeserialization="True"}}
+
+    @typing.overload
+    def _{{operationId}}_oapg(
+    {{> endpoint_args isOverload=true skipDeserialization="null"}}
+
+    def _{{operationId}}_oapg(
+    {{> endpoint_args isOverload=false skipDeserialization="null"}}
```

Comment:
There should be 3 signatures here:
- skipDeserialization True, overload with .. and return type hint
- skipDeserialization False, overload with .. and return type hint
- skipDeserialization null, non-overload with no response type hint
  - the skip_deserialization input for this should be `skip_deserialization: bool = False`

Can you update the template to do that?
Also, I think that you can use the True/False values in skipDeserialization to know that an overload signature is being generated and omit passing in isOverload entirely.
Have you checked the generated code in pycharm to make sure that the type hints are working correctly?